dolphin/task_manager/tasks/task.py

```diff
@@ -0,0 +1,133 @@
+        except AttributeError as e:
+            LOG.error(e)
+            raise exception.DolphinException(e)
+        except Exception as e:
+            msg = _('Failed to update storage entry in DB: {0}'
+                    .format(e))
+            LOG.error(msg)
+            raise exception.DolphinException(msg)
+        LOG.info("Syncing storage successful!!!")
+
+    def remove(self):
+        pass
+
+
+class StoragePoolTask(StorageResourceTask):
+    def __init__(self, context, storage_id):
+        super(StoragePoolTask, self).__init__(context, storage_id)
+
+    def _get_delete_list(self, storage_pools, db_pools):
+        """
+        :param storage_pools:
+        :param db_pools:
+        :return: prepare the delete_list. It will contain only those pool id
+        which is present in db but NOT in current list of storage pools
+        """
+        delete_list = copy.copy(db_pools)
+        [delete_list.remove(d_pool) for s_pool in storage_pools
+         for d_pool in db_pools if s_pool.get('original_id') == d_pool.get(
+            'original_id')]
+        return delete_list
+
+    def _get_add_list(self, storage_pools, db_pools):
+        """
+        :param storage_pools:
+        :param db_pools:
+        :return: prepare the add_list. It will contain only those pool id
+        which is NOT present in db but in current list of storage pools
+        """
+        add_list = copy.copy(storage_pools)
+        [add_list.remove(s_pool) for s_pool in storage_pools
+         for d_pool in db_pools if s_pool.get('original_id') == d_pool.get(
+            'original_id')]
+        return add_list
+
+    def _get_update_list(self, storage_pools, db_pools):
+        """
+        :param storage_pools:
+        :param db_pools:
+        :return: prepare the update_list. It will contain only those pool id
+        which is present in db and current list of storage pools
+        """
+        update_list = []
+        [update_list.append(s_pool) for s_pool in storage_pools
+         for d_pool in db_pools if s_pool.get('original_id') == d_pool.get(
+            'original_id')]
+        return update_list
+
+    def sync(self):
+        """
+        :return:
+        """
+        LOG.info('Syncing pool for storage id:{0}'.format(self.storage_id))
+        try:
+            # collect the pools list from drivermanger and dtabase
+            storage_pools = self.driver.list_pools(self.context,
+                                                   self.storage_id)
+            db_pools = db.pool_get_all(self.context)
+
+            db.pools_delete(self.context, self._get_delete_list(storage_pools,
+                                                                db_pools))
+
+            db.pools_update_all(self.context, self._get_update_list(
+                storage_pools, db_pools))
+
+            db.pools_create_all(self.context, self._get_add_list(
+                storage_pools, db_pools))
```

Comment:
didn't see thees db implementation in this pr

Reply:
ohh.sorry. Added that file